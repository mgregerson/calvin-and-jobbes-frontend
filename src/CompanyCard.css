img {
  width: 50px;
}

.CompanyCard {
  margin-bottom: 1rem;
  max-height: 250px;
}

.CompanyCard:hover {
}

.CompanyCard.card {
  display: flex;
  flex-direction: column;
}
