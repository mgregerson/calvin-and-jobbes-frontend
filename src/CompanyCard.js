import "./CompanyCard.css";

function CompanyCard({ company }) {
  return (
    <div className="CompanyCard card">
      <div className="Company Card card-body">
        <div className="CompanyCard card-title">
          <h6 className="CompanyCard-company float-start">{company.name}</h6>
          {company.logoUrl !== null && (
            <img
              className="float-end ms-5"
              src={`${company.logoUrl}`}
              alt={company.name}
            />
          )}
        </div>
        <p>
          <small className="CompanyCard-description float-start">
            {company.description}
          </small>
        </p>
      </div>
    </div>
  );
}

export default CompanyCard;
