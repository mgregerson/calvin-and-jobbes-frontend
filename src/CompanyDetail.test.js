import { render } from "@testing-library/react";
import CompanyCard from "./CompanyCard";

const TEST_COMPANY_LOGO = {
  handle: "APP",
  name: "Apple",
  description: "Company with computers",
  numEmployees: 400,
  logoUrl: "logos/logo1.png",
  jobs: [
    {
      id: 91,
      title: "Paramedic",
      salary: 122000,
      equity: "0.047",
    },
  ],
};

const TEST_COMPANY_NO_LOGO = {
  handle: "APP",
  name: "Apple",
  description: "Company with computers",
  numEmployees: 400,
  logoUrl: null,
  jobs: [
    {
      id: 91,
      title: "Paramedic",
      salary: 122000,
      equity: "0.047",
    },
  ],
};

test("renders without crashing", function () {
  render(<CompanyCard company={TEST_COMPANY_LOGO} />);
});

test("matches snapshot", function () {
  const { container } = render(<CompanyCard company={TEST_COMPANY_LOGO} />);
  expect(container).toMatchSnapshot();
});

test("puts info on page if there is a logo", () => {
  const { container } = render(<CompanyCard company={TEST_COMPANY_LOGO} />);

  expect(container.querySelector(".CompanyCard-company").innerHTML).toEqual(
    "Apple"
  );
  expect(container.querySelector(".CompanyCard-description").innerHTML).toEqual(
    "Company with computers"
  );
  expect(container.querySelector(".CompanyCard-img")).toBeInTheDocument();
});

// Renders if company has no logo

test("puts info on the page if there is no logo", function () {
  const { container } = render(<CompanyCard company={TEST_COMPANY_NO_LOGO} />);

  expect(container.querySelector(".CompanyCard-company").innerHTML).toEqual(
    "Apple"
  );
  expect(container.querySelector(".CompanyCard-description").innerHTML).toEqual(
    "Company with computers"
  );
  expect(container.querySelector(".CompanyCard-img")).not.toBeInTheDocument();
});
