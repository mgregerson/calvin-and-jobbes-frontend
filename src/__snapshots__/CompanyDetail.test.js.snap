// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <a
    class="CompanyCard card"
    href="companies/APP"
  >
    <div
      class="Company Card card-body"
    >
      <div
        class="CompanyCard card-title"
      >
        <h6
          class="CompanyCard-company"
        >
          Apple
        </h6>
        <img
          alt="Apple"
          class="CompanyCard-img float-end ms-5"
          src="logos/logo1.png"
        />
      </div>
      <small
        class="CompanyCard-description float-start"
      >
        Company with computers
      </small>
    </div>
  </a>
</div>
`;
